src/Network_Equivalences-Communication.thy

```diff
@@ -242,6 +1339,37 @@ lemma adapted_after_bidirectional_bridge:
   shows "(A \<leftrightarrow> B) \<guillemotleft> \<E> = A \<guillemotleft> \<E> \<leftrightarrow> B \<guillemotleft> \<E>"
   by (simp only: bidirectional_bridge_def adapted_after_parallel adapted_after_unidirectional_bridge)
 
+lemma bidirectional_bridge_forward_transition:
+  shows "A \<leftrightarrow> B \<rightarrow>\<^sub>s\<lparr>A \<triangleright> \<star>\<^bsup>n\<^esup> X\<rparr> ((B \<guillemotleft> suffix n \<triangleleft> X \<parallel> \<zero>) \<parallel> (A \<rightarrow> B) \<guillemotleft> suffix n) \<parallel> (B \<rightarrow> A) \<guillemotleft> suffix n"
+  unfolding bidirectional_bridge_def
+  by (intro unidirectional_bridge_transition parallel_left_io)
+
+lemma bidirectional_bridge_backward_transition:
+  shows "A \<leftrightarrow> B \<rightarrow>\<^sub>s\<lparr>B \<triangleright> \<star>\<^bsup>n\<^esup> X\<rparr> (A \<rightarrow> B) \<guillemotleft> suffix n \<parallel> ((A \<guillemotleft> suffix n \<triangleleft> X \<parallel> \<zero>) \<parallel> (B \<rightarrow> A) \<guillemotleft> suffix n)"
```

Comment:
The statement of this lemma should be `A ↔ B →⇩s⦇B ▹ ⋆⇗n⇖ X⦈ (A « suffix n ◃ X ∥ 𝟬) ∥ (A ↔ B) « suffix n`, for the reasons stated above.